pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionFactory.java

```diff
@@ -143,6 +143,31 @@ public static AggregationFunction getAggregationFunction(FunctionContext functio
             // PercentileRawTDigestMV
             return new PercentileRawTDigestMVAggregationFunction(firstArgument, percentile);
           }
+        } else if (numArguments == 3) {
+          // Triple arguments percentile (e.g. percentileTDigest(bar, 95, 1000), etc.) where the
+          // second argument is a decimal number from 0.0 to 100.0 and third argument is a decimal number indicating
+          // the compression_factor for the TDigest. This can only be used for TDigest type percentile functions to
+          // pass in a custom compression_factor. If the two argument version is used the default compression_factor
+          // of 100.0 is used.
+          // Have to use literal string because we need to cast int to double here.
+          double percentile = parsePercentileToDouble(arguments.get(1).getLiteral().getStringValue());
+          double compressionFactor = parseCompressionFactorToDouble(arguments.get(2).getLiteral().getStringValue());
```

Comment:
Should we validate compressionFactor > 0 or greater than some minimum ?

Reply:
done